src/main/java/org/openjdk/jextract/impl/TypedefBuilder.java

```diff
@@ -27,29 +27,21 @@
 
 import org.openjdk.jextract.Declaration;
 
+import java.util.List;
+
 public class TypedefBuilder extends ClassSourceBuilder {
     private final Declaration.Typedef typedefTree;
-    private final String superClass;
 
-    public TypedefBuilder(JavaSourceBuilder enclosing,
-        Declaration.Typedef typedefTree, String name, String superClass) {
-        super(enclosing, Kind.CLASS, name);
+    public TypedefBuilder(SourceFileBuilder builder, String className, String superClass,
+                          Declaration.Typedef typedefTree) {
+        super(builder, "public", Kind.CLASS, className, superClass, List.of());
         this.typedefTree = typedefTree;
-        this.superClass = superClass;
-    }
-
-    @Override
-    String superClass() {
-        return superClass;
     }
 
-    @Override
-    void classDeclBegin() {
+    void generate() {
```

Comment:
This method seems only to be called after we create a new `TypedefBuilder`. Same is true for `FunctionalInterfaceBuilder`. It's also not a special method, just something that clients need to remember to call.
I'm wondering if this patch isn't really trying to transition away from the "building" metaphore - e.g. what you want is to create a new "typedef", "functional interface" object, and add it to some list of generated objects (so that their code will be all co-generated at the end). Having the client to call `generate` explicitly so that the string buffer is filled seems redundant. Either the string buffer is filled at construction (eagerly), or it is filled when creating the java file object (lazily).

Reply:
Not suggesting to remove those builders. But I think the client saying `new FunctionalInterfaceBuilder(...)` should be enough.

Reply:
Okay. In that case I'd like to go with the static `generate` method. In terms of contract surface that is the same, but it allows giving a name to the code generation 'side effect' as well.